### ISSUE TITLE
V6 - Field does not rerender when its sync error changes

### ISSUE THREAD

**Issue by q42jaap** [erikras/redux-form#1024]
I have an array editor, and a validation function.

When I make changes in 1 field, my validation function runs and based on the value of field 1, the validation for field 2 might change (it's not passwords comparison).
When I make changes to field 1, validation is updated only for that field, but when the validation function removes an error for field 2, that error is not removed in the UI.

**Comment by ooflorent** [comment]
On v6, updating a field does not rerender other fields (unlike v6). This might be what's causing your issue. Could you add some code to diagnose what's happening here?

**Comment by imjared** [comment]
> On v6, updating a field does not rerender other fields (unlike v6)

@ooflorent this is the first I'm seeing of this (i'm guessing you meant "unlike v5"). Do you have more info (issues, docs, etc.) I could look into? I feel like I've looked through the docs, especially the migration guide, but I'm not seeing much else on this issue

**Comment by erikras** [comment]
I understand what @q42jaap means. This is a bug.

It's not particularly related to arrays. If you had a validation function like:
```js
const validate = values => {
  const errors = {}
  if (values.newPassword !== values.confirmNewPassword) {
    errors.confirmNewPassword = 'Must match!'
  }
  return errors
}
```
And then you edited the `newPassword` field in such a way that it added or removed the error on `confirmNewPassword`, the `<Field name="confirmNewPassword"/>` won't be rerendered....but it _should_ be!

@clayne11 is currently working on some changes (see #908) that will result in the sync validation errors being in the Redux store, which will make fixing this bug trivial.

**Comment by tnrich** [comment]
I'm seeing an issue where other Fields within a FieldArray aren't re-rendered when the field array has new elements inserted/removed. Is this related to this issue or should I make a new one?

Thanks!

**Comment by erikras** [comment]
> Is this related to this issue or should I make a new one?

New one. 😦

**Comment by erikras** [comment]
This is fixed. I will comment again when the fix is published.

**Comment by tnrich** [comment]
Thanks @erikras ! No pressure, but what's the eta on the new release?

**Comment by erikras** [comment]
Published in [`v6.0.0-rc.1`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-rc.1).
